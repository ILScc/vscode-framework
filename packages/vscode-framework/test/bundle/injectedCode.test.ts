/// <reference types="jest" />

import fs from 'fs'
import { join } from 'path'

// the main purpose of these tests is to see that bundler isn't misbehaving
// with this test it's relatively safe to keep esbuild up-to-date
// for example esbuild would remove declarations as unused etc in one of the future releases
test('Injected console code remains the same', async () => {
    const injectedCode = await fs.promises.readFile(
        join(__dirname, '../../build/cli/esbuild/consoleInject.js'),
        'utf-8',
    )
    expect(injectedCode.split('\n').slice(1).join('\n')).toMatchSnapshot()
})
