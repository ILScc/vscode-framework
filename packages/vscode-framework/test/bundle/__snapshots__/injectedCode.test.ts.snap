// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Injected console code remains the same 1`] = `
"const VSCODE_FRAMEWORK_OUTPUT = {
  oldConsole: globalThis.console,
  channel: void 0,
  consoleTimeFormatter: new Intl.DateTimeFormat(\\"en-US\\", {
    hour: \\"2-digit\\",
    minute: \\"2-digit\\",
    second: \\"2-digit\\",
    fractionalSecondDigits: 3,
    hour12: false
  }),
  isDebugEnabled: false,
  currentLevel: void 0,
  appendOutput(message) {
    const levelString = this.currentLevel ? \` [\${this.currentLevel}]\` : \\"\\";
    this.channel.append(\`[\${this.consoleTimeFormatter.format(new Date())}]\${levelString} \${message}\`);
    this.currentLevel = void 0;
  },
  newConsole: new globalThis.console.Console(...(() => {
    const { PassThrough } = require(\\"stream\\");
    const stdout = new PassThrough();
    stdout.write = (message) => VSCODE_FRAMEWORK_OUTPUT.appendOutput(message);
    const stderr = new PassThrough();
    stderr.write = (message) => {
      VSCODE_FRAMEWORK_OUTPUT.currentLevel = \\"error\\";
      VSCODE_FRAMEWORK_OUTPUT.appendOutput(message);
    };
    return [stdout, stderr];
  })())
};
const process = {
  ...global.process,
  stderr: {
    ...global.process.stderr,
    write(string, cb) {
      VSCODE_FRAMEWORK_OUTPUT.channel.append(string);
      cb == null ? void 0 : cb(null);
      return true;
    }
  }
};
const vscode_framework_set_debug_enabled = (isEnabled) => {
  VSCODE_FRAMEWORK_OUTPUT.isDebugEnabled = isEnabled;
};
const console = {
  ...VSCODE_FRAMEWORK_OUTPUT.oldConsole,
  ...VSCODE_FRAMEWORK_OUTPUT.newConsole,
  ...{
    debug(...data) {
      if (!VSCODE_FRAMEWORK_OUTPUT.isDebugEnabled)
        return;
      VSCODE_FRAMEWORK_OUTPUT.currentLevel = \\"debug\\";
      VSCODE_FRAMEWORK_OUTPUT.newConsole.log(...data);
    },
    warn(...data) {
      VSCODE_FRAMEWORK_OUTPUT.currentLevel = \\"warn\\";
      VSCODE_FRAMEWORK_OUTPUT.newConsole.log(...data);
    },
    log(...data) {
      VSCODE_FRAMEWORK_OUTPUT.currentLevel = \\"log\\";
      VSCODE_FRAMEWORK_OUTPUT.newConsole.log(...data);
    },
    clear() {
      VSCODE_FRAMEWORK_OUTPUT.channel.clear();
    },
    show() {
      VSCODE_FRAMEWORK_OUTPUT.channel.show();
    },
    hide() {
      VSCODE_FRAMEWORK_OUTPUT.channel.hide();
    }
  }
};
"
`;
